model.py

```diff
@@ -4,12 +4,16 @@
 from dataclasses import dataclass
 import csv
 
-@dataclass
+@dataclass (frozen=True)
 class Criterion:
     name: str
     description: str
     scores: List[str]
-
+    
+    def __hash__(self):
+        # Convert list to tuple for hashing purposes
+        return hash((self.name, self.description, tuple(self.scores)))
+    
```

Comment:
You don't have to write your own hash function if you make the Criterion `scores` field a tuple (i.e. `scores: tuple[str, ...]`). You will, however, have to update the `load_criteria` function to pass a tuple rather than a list when you create Criterion objects.

Reply:
I've made the changes, but I've been wondering about self-defined hash, I originally thought it would be cleaner to use it so I could avoid changing existing code, but is it unnecessary in this case?

Reply:
I see, thank you for explaining!